lib/Relation/NestedRelation.php

```diff
@@ -0,0 +1,179 @@
+<?php
+namespace Spot\Relation;
+
+use Spot\EntityInterface;
+use Spot\Entity\Collection;
+
+/**
+ * NestedRelation
+ *
+ * Used for eager-loading multilevel relations
+ *
+ * @package Spot
+ */
+class NestedRelation extends RelationAbstract
```

Comment:
There are no tests for this new behavior. We need to see some tests and possibly some new test entities also with the proper setup done (as shown in your code sample in this PR). I cannot merge anything new without new tests also that demonstrate the behavior.

Reply:
Sure, I will add some tests.